gnocchi/rest/aggregates/operations.py

```diff
@@ -150,6 +154,47 @@ def handle_binary_operator(nodes, granularity, timestamps,
     return granularity, timestamps, values, is_aggregated
 
 
+def handle_ternary_operator(nodes, granularity, timestamps,
+                            initial_values, is_aggregated, references):
+    op = nodes[0]
+    g1, t1, v1, is_a1 = evaluate(nodes[1], granularity, timestamps,
+                                 initial_values, is_aggregated, references)
+    g2, t2, v2, is_a2 = evaluate(nodes[2], granularity, timestamps,
+                                 initial_values, is_aggregated, references)
+    if len(nodes) > 3:
+        g3, t3, v3, is_a3 = evaluate(nodes[3], granularity, timestamps,
+                                     initial_values, is_aggregated, references)
+    else:
+        g3, t3, v3, is_a3 = g2, t2, None, is_a2
+
+    is_aggregated = is_a1 or is_a2 or is_a3
+    if isinstance(v1, numpy.ndarray) and isinstance(v2, numpy.ndarray)\
+            and isinstance(v3, numpy.ndarray):
+        if not numpy.array_equal(t1, t2) or g1 != g2:
+            if not numpy.array_equal(t2, t3) or g2 != g3:
+                raise exceptions.OperandsMismatch(
+                    references,
+                    "Can't compute timeseries with different "
+                    "granularity %s <> %s <> %s"
+                    % (nodes[1], nodes[2], nodes[3]))
+        timestamps = t1
+        granularity = g1
+        is_aggregated = True
+
+    elif isinstance(v2, numpy.ndarray):
+        timestamps = t2
+        granularity = g2
+    elif isinstance(v3, numpy.ndarray):
+        timestamps = t3
+        granularity = g3
+    else:
+        timestamps = t1
+        granularity = g1
```

Comment:
sort of torn regarding the code from lines 170 to 192... None of it actually is applicable to numpy.clip and i'm not sure what other operations would actually require it. thoughts?

specifically, there is function specific conditions. for example, if nodes[2] or nodes[3] aren't the same length as nodes[1], this will throw an error.

Reply:
I agree that nodes[2&3] aren't the same length as nodes[1] then why do we do the same sort of verification in handle_binary_operator as nodes[1] and nodes[2] don't have same length as well.

Reply:
yeah, you're right. the above code does check the shape.

i guess the only part that is irrelevant is the granularity check `g1 != g2` and  `g2 != g3`. granularity should not really matter but i guess this code is still relevant.